src/algorithms/synthesis/syrec_synthesis.cpp

```diff
@@ -0,0 +1,1447 @@
+        } else {
+            const_line = add_line_to_circuit(_circ, (std::string("const_") + std::to_string(value)), "garbage", value, true);
+        }
+
+        return const_line;
+    }
+
+    bool standard_syrec_synthesizer::get_constant_lines(unsigned bitwidth, unsigned value, std::vector<unsigned>& lines) {
+        boost::dynamic_bitset<> number(bitwidth, value);
+
+        for (unsigned i = 0u; i < bitwidth; ++i) {
+            lines.emplace_back(get_constant_line(number.test(i)));
+        }
+
+        return true;
+    }
+
+    void standard_syrec_synthesizer::add_variable(circuit& circ, const std::vector<unsigned>& dimensions, const applications::variable::ptr& var,
+                                                  constant _constant, bool _garbage, const std::string& arraystr) {
+        if (dimensions.empty()) {
+            for (unsigned i = 0u; i < var->bitwidth(); ++i) {
+                std::string name = boost::str(boost::format(variable_name_format) % var->name() % i % arraystr);
+                add_line_to_circuit(circ, name, name, _constant, _garbage);
+            }
+        } else {
+            unsigned              len = dimensions.front();
+            std::vector<unsigned> new_dimensions(dimensions.begin() + 1u, dimensions.end());
+
+            for (unsigned i = 0u; i < len; ++i) {
+                add_variable(circ, new_dimensions, var, _constant, _garbage, arraystr + "[" + std::to_string(i) + "]");
+            }
+        }
+    }
+
+    void standard_syrec_synthesizer::add_variables(circuit& circ, const applications::variable::vec& variables) {
+        for (const auto& var: variables) {
+            // entry in var lines map
+            _var_lines.insert(std::make_pair(var, circ.lines()));
+
+            // types of constant and garbage
+            constant _constant = (var->type() == applications::variable::out || var->type() == applications::variable::wire) ? constant(false) : constant();
+            bool     _garbage  = (var->type() == applications::variable::in || var->type() == applications::variable::wire);
+
+            add_variable(circ, var->dimensions(), var, _constant, _garbage, std::string());
+        }
+    }
+
+    bool syrec_synthesis(circuit& circ, const applications::program& program, const properties::ptr& settings, const properties::ptr& statistics) {
+        // Settings parsing
+        auto variable_name_format  = get<std::string>(settings, "variable_name_format", "%1$s%3$s.%2$d");
+        auto main_module           = get<std::string>(settings, "main_module", std::string());
+        auto statement_synthesizer = get<standard_syrec_synthesizer>(settings, "statement_synthesizer", standard_syrec_synthesizer(circ, program));
+
+        statement_synthesizer.set_settings(settings);
+
+        // Run-time measuring
+        timer<properties_timer> t;
+
+        if (statistics) {
+            properties_timer rt(statistics);
+            t.start(rt);
+        }
```

Comment:
this starts a timer, but the timer is never stopped.
this doesn't seem adequate for measuring the synthesis runtime.